hearth/lib/hearth/request.rb

```diff
@@ -19,5 +18,11 @@ def initialize(uri: URI(''), body: StringIO.new)
 
     # @return [IO]
     attr_accessor :body
+
+    # @api private
+    def initialize_copy(other)
+      @uri = other.uri.dup
+      # body should not be copied
```

Comment:
But should it be rewound (if respond_to rewind)?

Reply:
That's done in retry. This is for dup / copy. But I removed this in favor of signer reset.